bluepyopt/electrical/parameters.py

```diff
@@ -93,6 +93,10 @@ def check_bounds(self):
                     (self.name, self._value, str(self.lower_bound),
                      str(self.upper_bound)))
 
+    def destroy(self):
```

Comment:
Ok, does that apply only to parameters?

For example, bluepyopt/electrical/protocols.py calls destroy on all stimulus & recordings.  I moved their destroys up to the base class, so knows they have to implement it, otherwise there's no hint as to which methods are needed to be a valid stimulus.

The alternative, imo, is to use ABCs such that if a method isn't defined, but is needed, then the class can't be instantiated.